Add docs for Pixie Plugin System (#154)
pixie-labs/pixie-docs: content/en/05-reference/03-plugins/01-plugin-system.md

---
title: "Pixie Plugin System"
metaTitle: "Reference | Plugins | Pixie Plugin System "
metaDescription: "Pixie's Plugin System"
order: 1 
---

> Pixie's core functionality focuses on providing a real-time debugging platform. Integrate with other tools using the Pixie Plugin system to enable capabilities like longterm data retention or alerting.

Reviewer: ```suggestion
Pixie's primary focus is to build a real-time debugging platform, not a full-fledged observability solution. Pixie provides a plugin system to integrate with external tools to enable capabilities like longterm data retention and alerting.
```

> - *Longterm Data Retention*: Pixie does not guarantee data storage for over 24 hours. Leverage an external datastore for longterm data retention by sending Pixie data in the [OpenTelemetry](https://opentelemetry.io/) format. Future support will be added for querying longterm data from within the Pixie UI, allowing you to use PxL and Pixie's versatile live views.

Reviewer: ```suggestion
- *Longterm Data Retention*: Pixie only retains up to 24 hours of data. Leverage an external datastore for longterm data retention by sending Pixie data in the [OpenTelemetry](https://opentelemetry.io/) format. Future support will be added for querying longterm data from within the Pixie UI, allowing you to use PxL and Pixie's versatile live views.
```

Reviewer: Can you link "Longterm Data Retention" to the header below? (I can't remember the exact format - check the FAQ page).

- *Alerts* (Coming Soon!): Power alerts using Pixie's rich dataset, all configurable from within Pixie's UI.

## Enabling a Plugin

Plugins can be configured from within the Admin UI. 

1. Open the Live UI and navigate to the Plugins tab on the Admin page (`/admin/plugins`).
2. Find the plugin you wish to enable and click the toggle to enable/disable the plugin.

> 3. Expand the plugin row to specify the necessary configuration values. The required set of fields differs per plugin provider, and usually contains information such as API keys which are used for authentication.

Reviewer: Link to /reference/admin/api-keys.

Author: This is the API key for the plugin provider, not ours.

## Longterm Data Retention

Enable a plugin which offers longterm data retention capabilities to send Pixie data to an external datastore. This plugin allows you to configure PxL scripts to export data at regularly scheduled intervals. This only currently supports exporting data through the [OpenTelemetry](https://opentelemetry.io/) format. 

By default, the plugin provider has configured a set of preset scripts. These will automatically start running and exporting data from all clusters in the org as soon as the plugin is enabled. Users can also choose to export custom data by creating a custom export script.

### Configuring Preset Export Scripts

> 1. [Enable a plugin provider](#enable-a-plugin) which supports longterm data retention. 

Reviewer: link is broken

2. Open the Live UI and navigate to the Data Export page in the sidebar (`/configure-data-export`).
3. All preset scripts are listed under `<Plugin Provider> Scripts`.
4. Click the toggle to enable/disable the export from the preset script.
5. Click the gear icon to configure the preset script. Name, description, and PxL is uneditable. However, users may choose which clusters to export from and how frequently the script is run.
6. Click "Save" to update the script with your changes.

### Creating Custom Export Scripts

> 1. [Enable a plugin provider](#enable-a-plugin) which supports longterm data retention. 

Reviewer: link is broken
